### PR TITLE
CCTool Unpack

### PR DESCRIPTION
**Pull request by ldelossa**
This PR implements a command to unpack a particular image reference to your temp directory.
I keep having to do this when detection fails for a particular image and this slipstreams that process

Signed-off-by: ldelossa <ldelossa@redhat.com>

### CODE REVIEW

File: cmd/cctool/unpack.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+func Unpack(cmd context.Context, cfg *commonConfig, args []string) error {
+	cmdcfg := unpackConfig{}
+	fs := flag.NewFlagSet("cctool unpack", flag.ExitOnError)
+	fs.DurationVar(&cmdcfg.timeout, "timeout", 5*time.Minute, "timeout for successful http responses")
+	fs.Usage = func() {
+		out := fs.Output()
+		fmt.Fprintf(out, "Usage:\n")
+		fmt.Fprintf(out, "\tcctool unpack <image-ref>\n")
+		fmt.Fprintf(out, "Arguments:\n")
+		fmt.Fprintf(out, "\timage-ref: a reference to an image and it's repository\n\n")
```
Comment (hdonnay) [inline cmd/cctool/unpack.go]: s/'//

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@ vendor
 *.swp
 book
 cctool
+!cctool/
```
Comment (hdonnay) [inline .gitignore]: Why is this needed?

File: cmd/cctool/main.go [hunk 3]
```diff
@@ -14,6 +14,7 @@ import (
 var cleanup sync.WaitGroup
 
 type commonConfig struct {
+	cleanup *sync.WaitGroup
```
Comment (hdonnay) [inline cmd/cctool/main.go]: removing the pointer means the zero-value of commonConfig is OK to use

File: cmd/cctool/unpack.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	}
+	log.Printf("successfully fetched layers")
+
+	// create a tmp dir we will unpack layers to
+	td, err := ioutil.TempDir("", "cctool-unpack-")
+
+	log.Printf("exacting layers into tmp dir: %v.", td)
+	for i, layer := range m.Layers {
+		dirName := filepath.Base(layer.Local())
+		dir := filepath.Join(td, strconv.Itoa(i)+"-"+dirName)
```
Comment (hdonnay) [inline cmd/cctool/unpack.go]: I think it'd be easier to use the layer digest instead of grabbing the local file name.

File: .gitignore (hunk 2 above)
Comment (ldelossa) [reply .gitignore]: i assume the initial intent was to not include cctool binary. But by doing this that regex also matches the dir name and ignores any modifications to files children of /cmd/cctool

File: cmd/cctool/main.go (hunk 3 above)
Comment (ldelossa) [reply cmd/cctool/main.go]: Is it okay to copy the wait group between method calls? I defaulted to the sync.Mutex rules where you dont want to copy the mutex

File: .gitignore (hunk 2 above)
Comment (hdonnay) [reply .gitignore]: ah yeah whoops

File: cmd/cctool/main.go (hunk 3 above)
Comment (hdonnay) [reply cmd/cctool/main.go]: the commonConfig is passed by a pointer to the subcommand, so it should be fine.

Comment (ldelossa) [reply cmd/cctool/main.go]: hmm, imo the usage of the cleanup waitgroup shouldn't break by you deciding to pass a reference or copy of the containing struct to a method. that seems like a hard to catch bug?

Comment (hdonnay) [reply cmd/cctool/main.go]: you'd have to purposefully copy the struct, as it comes in as a pointer

Comment (ldelossa) [reply cmd/cctool/main.go]: Well I get that, but if you wrote a function that *didn't* to that, you'd break the usage of commonConfig, which is weird.

File: layer.go [hunk 5]
```diff
@@ -28,6 +28,10 @@ func (l *Layer) SetLocal(f string) error {
 	return nil
 }
 
+func (l *Layer) Local() string {
+	return l.localPath
+}
+
```
Comment (hdonnay) [inline layer.go:34]:
```suggestion
```
